src/sqlfluff/rules/L009.py

```diff
@@ -119,8 +115,20 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                 ],
             )
         else:
-            # There are excess newlines so delete all bar one.
-            return LintResult(
-                anchor=context.segment,
-                fixes=[LintFix.delete(d) for d in trailing_newlines[1:]],
+            # There are one or more trailing newlines in templated space.
+            # For any excess newlines, translate to "raw" space to determine if
+            # they are literal (rather than templated). Delete any extras.
+            # (Note that 'trailing_newlines' is ordered in reverse, i.e. from
+            # the end of the file *backwards*.)
+            extra_newlines = trailing_newlines[1:].select(
+                loop_while=lambda seg: sp.raw_slices(seg, context.templated_file).all(
+                    rsp.is_slice_type("literal")
+                )
```

Comment:
@barrywhart you're missing a case here when trailing_newlines[0] is also a template segment:

```sql
SELECT foo
FROM bar

{{ "\n\n" }}$
```
i.e. we've detected there are newlines, but the last one is inside the template so we need to add another one below.

Reply:
test case to satisfy
```yml
test_fail_templated_plus_raw_newlines:
  fail_str: "{{ '\n\n' }}"
  fix_str: "{{ '\n\n' }}\n"
```

Reply:
Good point! I'll open another PR to add this.